build-docker-image.sh

```diff
@@ -35,9 +39,19 @@ if [ -x "scripts/docker_build.sh" ]; then
 else
     docker_build="docker build"
 fi
+
+declare -a build_args
+if grep -q '^ARG KATSDPDOCKERBASE_REGISTRY' Dockerfile; then
+    build_args+=(--build-arg "KATSDPDOCKERBASE_REGISTRY=$DOCKER_REGISTRY")
```

Comment:
This presumably overrides whatever registry has been specified in the Dockerfile. Is there point in allowing a Jenkins build without $DOCKER_REGISTRY set and it then uses the default from the Dockerfile ?

Reply:
With this change the DOCKER_REGISTRY Jenkins setting is now describing two things:
- where to **read** docker-base images from, in Dockerfiles that support it (new)
- where to **write** the built image (already the case)

We could split the functions, but do you have a use case in mind? If not then I'd rather wait until we need to cross that bridge.

Reply:
Thanks for the explanation - no particular use case in mind (as yet)...